File: src/ExpoPayModule.ts
import { NativeModule, requireNativeModule } from 'expo';

import { ExpoPayModuleEvents } from './ExpoPay.types';

declare class ExpoPayModule extends NativeModule<ExpoPayModuleEvents> {
  PI: number;
  hello(): string;
  setValueAsync(value: string): Promise<void>;
}

export default requireNativeModule<ExpoPayModule>('ExpoPay');

File: android/src/main/java/expo/modules/pay/ExpoPayModule.kt
package expo.modules.pay

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoPayModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoPay")

    Events("onChange")

    Constant("PI") {
      Math.PI
    }

    Function("hello") {
      "Hello world! 👋"
    }

    AsyncFunction("setValueAsync") { value: String ->
      sendEvent("onChange", mapOf(
        "value" to value
      ))
    }

    View(ExpoPayView::class) {
    }
  }
}
